packages/flutter/lib/src/painting/decoration_image.dart

```diff
@@ -295,6 +295,11 @@ class DecorationImagePainter {
   void _handleImage(ImageInfo value, bool synchronousCall) {
     if (_image == value)
       return;
+    if (_image != null && _image!.isCloneOf(value)) {
+      value.image.dispose();
```

Comment:
It's really sketchy that we're disposing something inside an object. That doesn't seem right (intuitive) at all.